src/app-layer-parser.c

```diff
@@ -1770,21 +1771,7 @@ void AppLayerParserRegisterProtocolParsers(void)
     RegisterRdpParsers();
     RegisterHTTP2Parsers();
     rs_telnet_register_parser();
-
-    /** IMAP */
-    AppLayerProtoDetectRegisterProtocol(ALPROTO_IMAP, "imap");
-    if (AppLayerProtoDetectConfProtoDetectionEnabled("tcp", "imap")) {
-        if (AppLayerProtoDetectPMRegisterPatternCS(IPPROTO_TCP, ALPROTO_IMAP,
-                                  "1|20|capability", 12, 0, STREAM_TOSERVER) < 0)
```

Comment:
Is this good to remove ?

Reply:
-  as per [rfc-9051](https://datatracker.ietf.org/doc/html/rfc9051#name-commands-and-responses) there should be UID + command,  UID length is not fixed.